### ISSUE TITLE
pickle.unpicklingError: invalid load key, 'v'; Driver options for Linux system;  Dataset Links

### ISSUE THREAD

**Issue by imethanlee** [lindsey98/PhishIntention#2]
Hi,

I came across an error (after installing all the required pakages) when I ran the code in the [Use it as a packgake](https://github.com/imethanlee/PhishIntention#use-it-as-a-package) section (Run in python to test a single site). The specific error screenshot is
![issue](https://user-images.githubusercontent.com/36200682/167568568-4cb07464-d250-49cc-bd2a-6e1bf52df5fc.jpg)
What could be the possible reasons for this error?

BTW, you may want to know some package versions in my environment:
python 3.7
pytorch 1.10.0
Detectron2 for CUDA 11.3 and torch 1.10

Thanks for your help.

**Comment by lindsey98** [comment]
Hi,
  Can you please try to download the models directly from here: https://drive.google.com/drive/folders/1XGiLfIeSHwoeoXEpMXhMR4M2tkj3pErJ?usp=sharing. 
  And then move it to your /home/../miniconda3/envs/phish/lib/python3.7/site-packages/phishintention (basically where you installed the PhishIntention).
  Make sure that it follows this structure:
  ```
  Under phishintention/phishintention/src
    |___ AWL_detector_utils/: scripts for abstract layout detector 
        |__ output/
            |__ website_lr0.001/
                |__ model_final.pth
    |___ crp_classifier_utils/: scripts for CRP classifier
            |__ output/
                |__ Increase_resolution_lr0.005/
                    |__ BiT-M-R50x1V2_0.005.pth.tar
    |___ crp_locator_utils/: scripts for CRP locator 
        |__ login_finder/
            |__ output/
                |__ lr0.001_finetune/
                    |__ model_final.pth
    |___ OCR_siamese_utils/: scripts for OCR-aided Siamese
        |__ demo_downgrade.pth.tar
        |__ output/
            |__ targetlist_lr0.01/
                |__ bit.pth.tar
```

**Comment by imethanlee** [comment]
Thanks for the reply. Seems working now with your provided files. I think the problem is that Github somehow corrupts the big files (e.g., those .pth files), so replacing with the files you provided can make this work. 

There is actually a new problem coming up. Here is the screenshot,
![image](https://user-images.githubusercontent.com/36200682/167615374-26082e2f-f14a-413e-92af-1fc0bad91a2a.png)
How should I make this 'Chrome' thing work?

Thanks. 

UPDATE:
I am using a remote ubuntu server.

**Comment by lindsey98** [comment]
Hi,  
   It seems that your machine couldn't locate chrome binary under the default location /usr/bin/google-chrome. Here are some discussions: https://stackoverflow.com/questions/50138615/webdriverexception-unknown-error-cannot-find-chrome-binary-error-with-selenium
   Please have a check.

**Comment by imethanlee** [comment]
Ah, looks like I don't have a Chrome installed in the remote machine. I will try to install it. Thanks for the help.

**Comment by imethanlee** [comment]
Seems working after installing Chrome. For linux environment, some additional lines may need to be added in `phishintention_config.py`, which is 

```
options.add_argument('--no-sandbox')
options.add_argument('--disable-dev-shm-usage')
options.add_argument('--disable-gpu')
options.add_argument('--headless')
```

Here is the screenshot of the output message (of the Use it as a packgake/Run in python to test a single site). Would like to check if it is a normal output? There is a 'timeout' so I am not sure whether it is supposed to appear. 

![image](https://user-images.githubusercontent.com/36200682/167800652-6ff687c7-6835-479b-b031-929069458f0f.png)

**Comment by lindsey98** [comment]
Hi,
   Sometimes even normal websites may have this error, so it is expected.
   But to know what exactly happens here, you need to switch to non-headless mode if possible.

**Comment by lindsey98** [comment]
But it seems that this webpage is already a credential-requiring-page, and the previous module CRP classifier didn't recognise it. So it is a false negative case made by PhishIntention. 
The motivation of PhishIntention is to avoid false positives made by Phishpedia (which only has a logo matching module), so the additional credential-requiring classification model may introduce some false negative problems. 
Would you like to share the site you are testing if possible? I am interested to investigate why it is not reported by CRP classifier.

**Comment by imethanlee** [comment]
Hi,
Sure. Thanks for your help. The website I test is exactly the same as the one you provided in the 'Use it as a packgake/Run in python to test a single site'. 
![image](https://user-images.githubusercontent.com/36200682/167862131-e76c5c24-a124-4dc5-8f42-a7e474d35c9e.png)

The only difference between my code and your code is the additional 4 lines mentioned in https://github.com/lindsey98/PhishIntention/issues/2#issuecomment-1123326467. Seems if I do not include these four lines, an error will occur. Here is a screenshot of the output message if I do not include these 4 lines. 

![image](https://user-images.githubusercontent.com/36200682/167862427-e84d4b82-e346-42f2-b9bb-e23427eba036.png)

**Comment by lindsey98** [comment]
Hi,
  Thanks for confirming. In that case, let's keep these 4 lines. 
  By the way, the website was a phishing and it has changed its content now (the current screenshot is shown below and this is the previous screenshot https://github.com/lindsey98/PhishIntention/blob/main/phishintention/datasets/test_sites/accounts.g.cdcde.com/shot.png). 

 If you want to test some updated and fresh phishing websites. I think you can find some URLs here https://openphish.com

![Screenshot 2022-05-11 at 10 03 02 PM](https://user-images.githubusercontent.com/51521323/167868657-399e7e2d-23e0-4bea-b77b-ae11b777dc0e.png)

**Comment by lindsey98** [comment]
Also, may I confirm you have the following outputs printed when you reach line 15-18: 

A target is reported by siamese, enter CRP classifier
Already a CRP, continue
Phishing is found!
Phishing (1) or Benign (0) ? 1
What is its targeted brand if it is a phishing ? Google
What is the siamese matching confidence ? 0.9679305

**Comment by imethanlee** [comment]
Yes, I have the output you mentioned, except that my matching condifence is 0.9679289. I will have a try on other websites, and will reach you if other issues happen. Thanks for your help.

**Comment by lindsey98** [comment]
Thanks for checking. Actually, when you are testing you only need to run till line 21. The rest parts of the codes are the inner steps inside the test() function. I put them here in case someone wants to run individual steps. I am sorry for the confusion, I have updated the readme

**Comment by imethanlee** [comment]
Thanks. I would like to ask some questions about reproducing the results of the paper. Let's say 9010 webpages you mentioned section 5.1. Is this dataset available to the public? Another question is that to run the test, do we need to collect the screenshot, HTML code in advance, instead of running the test directly with the urls only?

**Comment by lindsey98** [comment]
Hi, yes the datasets are available, the dataset links are listed here: https://sites.google.com/view/phishintention/experiment-structure. 
Yes, you need to crawl them in advance, the folder should include info.txt (storing the URL), shot.png (screenshot), html.txt (page source). For small-scale crawling, you can use some common libraries like Selenium. For large-scale crawling, scrapy-splash is more efficient, we have released the crawling scripts for the phish-discovery experiment, [repo link](https://github.com/lindsey98/MyScrapy), you can have a check if you are interested.

**Comment by imethanlee** [comment]
Resolved.

**Comment by imethanlee** [comment]
To other users: 

The problem in https://github.com/lindsey98/PhishIntention/issues/2#issue-1230714197 can be easily solved by installing Git LFS (https://git-lfs.github.com/) to the machine you use. After installing Git LFS, the clone process will download all the files correctly. Or if you have cloned this repo, you can simpy run `git lfs pull` to update your files. 

The problem originates from the large files management by Github https://docs.github.com/en/repositories/working-with-files/managing-large-files. No need to replace any files mentioned in https://github.com/lindsey98/PhishIntention/issues/2#issuecomment-1122044102 after installing Git LFS actually.

**Comment by lindsey98** [comment]
Thanks for your suggestions!

**Comment by imethanlee** [comment]
Thanks for the authors' kind help for all the new issues!
